### DISCUSSION TITLE
Suggestion: I think we could make this even simpler?

### DISCUSSION THREAD

**Discussion by anasouardini** [pmndrs/jotai#1716] [Ideas]
I'm curious to see if we can write less code when importing/using an atom.

# what I think will be improved:
  - the atom is usable in the same component with a single line of code
  - the atom could be imported and used in another component also with single line

The defaults do not have to change, maybe it could be added as a second option. but you get the point.

``` js
import {atom} from 'jotai'

export default function myComponentA(){

  // the first attribute is the key that would be used to reference the atom
  const [counter, setCounter] = atom('myCounter', 1);
  const [title, setTitle] = atom('myTitle', 'atoms are great');

  return <h2>{title} - {counter}</h1>
}
```

Then we could use it like so:
```js
import {useAtom} from 'jotai'

export default function myComponentB(){

  // we can get the atom using it's previously specified key
  const [counter, setCounter] = useAtom('myCounter');

  const increaseGlobalCounter = ()=>{
    setCounter((prev)=>prev+1);
  }

  return <button onclick={increaseGlobalCounter}>{title}</button>
}
```

**Comment by dai-shi** [comment]
Thanks for a suggestion. 
There are actually many libs out there with string keys.  It's jotai's design decision to use atom references. 

Out of curiosity, what would happen if ComponentB is used before ComponentA?

**Reply by anasouardini** [reply]
totally understandable. I'll try to look for another lib.

well... the same would/could happen if you import a non existing atom.

but typically... `A` would be the parent if not grand-parent of `B`.

and since `A` and `B` would be kind of independent from each other to be mounted in a non synchronous way, The state/atom would have to be initialized in some common ancestor of the two.

if not, then a browser yelling at you is the best thing that could happen at that point imo.

**Reply by Samathingamajig** [reply]
Based on [a discussion with the OP on discord](https://discord.com/channels/966627436387266600/1068385452039618621) (in [theo's server](https://discord.gg/jBEuNF8ChA)), @segfaulty1 was looking for the DX of [zustand](https://github.com/pmndrs/zustand), another state management library made by [pmndrs](https://github.com/pmndrs) (just like this one)

edit: i had "the @segfaulty1" which i replaced to just "@segfaulty1" because i originally typed "the OP", remove the "OP" but forgot to remove the "the"
